src/cryptsetup/cryptsetup.c

```diff
@@ -2373,11 +2420,30 @@ static int run(int argc, char *argv[]) {
 
 /* since cryptsetup 2.7.0 (Jan 2024) */
 #if HAVE_CRYPT_SET_KEYRING_TO_LINK
-                        if (arg_link_key_description) {
+                        if (arg_link_keyring && arg_link_key_description)
                                 r = crypt_set_keyring_to_link(cd, arg_link_key_description, NULL, arg_link_key_type, arg_link_keyring);
-                                if (r < 0)
-                                        log_warning_errno(r, "Failed to set keyring or key description to link volume key in, ignoring: %m");
+                        else if (arg_link_keyring) {
+                                _cleanup_free_ char *key_description_1 = NULL, *key_description_2 = NULL;
+
+                                key_description_1 = strjoin("systemd-cryptsetup:vk-",
+                                                            strempty(crypt_get_uuid(cd)),
+                                                            "-id1");
+                                if (!key_description_1)
+                                        return log_oom();
+
+                                key_description_2 = strjoin("systemd-cryptsetup:vk-",
+                                                            strempty(crypt_get_uuid(cd)),
+                                                            "-id2");
+                                if (!key_description_2)
+                                        return log_oom();
+
+                                r = crypt_set_keyring_to_link(cd, key_description_1, NULL, arg_link_key_type, arg_link_keyring);
+                                /* use key_description_2 only if device requires two volume keys */
```

Comment:
hmm, what is this business with two volume keys btw? this is about reencryption? but we don't issue reencryption requests?

Reply:
Upon activation, the device may already be in 'in-reencryption' state. If that's the case there might be two different VKs needed to properly re-activate the device later.

Reply:
ok, got it. i'd still default to a name without "-id1" for the first key though, and only add some suffix to the 2nd

Reply:
To me, it seemed easier to look for a single prefix/format of the key in a keyring while searching for the keys. That's what users of this feature will do at some point after all devices gets activated.